src/core/protocol-authorization.ts

```diff
@@ -24,18 +24,18 @@ export class ProtocolAuthorization {
     tenant: string,
     incomingMessage: RecordsRead | RecordsWrite,
     requesterDid: string | undefined,
-    messageStore: MessageStore
+    storageController: StorageController
```

Comment:
This interesting to me. My interpretation of the original issue is that `StorageController` is the "better" abstraction for various handlers which made sense to me because instead of passing multiple interfaces all handlers need, define an abstraction that encapsulate those interfaces and simply pass one abstraction to the handlers.

However this PR seems to have taken one step further and elevated `StorageController` as THE abstraction that all code should interface with when it comes to all things store related.

Have not made up my mind yet on if this new model is better or worse, but it is clearly that in the case of protocol authorization here, it only needed messages to do its job, but is now given greater access to all stores unnecessarily, which goes against the generally accepted good programming principle of interface segregation.

Reply:
I agree with all of @thehenrytsai's points, except that I'd go a step further to say I definitely do not like turning `StorageController` into THE abstraction. `StorageController` shines best IMO as a util for common operations that involve multiple stores. In all other cases, it is better (more readable, explicit, modular) to pass the underlying store.

United storage into a single API does not come off as easier to understand or more wieldy than having distinct, separately-concerned storage interfaces.

Reply:
don't understand well enough to give a coherent response but happy to jump on a call to better understand